src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -484,10 +484,14 @@ def _rgw_add(self, realm_name, zone_name, num=1, hosts=None, inbuf=None):
             except ValueError as e:
                 msg = 'Failed to read JSON input: {}'.format(str(e)) + usage
                 return HandleCommandResult(-errno.EINVAL, stderr=msg)
-        rgw_spec = orchestrator.RGWSpec(
-            rgw_realm=realm_name,
-            rgw_zone=zone_name,
-            placement=orchestrator.PlacementSpec(hosts=hosts, count=num))
+        elif realm_name and zone_name:
```

Comment:
`zone_name is not None and realm_name is None` should work as well, right?

Reply:
Sure, `if variable:` covers both `None` and `''` both of which are illegal in this case. But sure, happy to change this if we only want to check for not None.